### PR TITLE
[Admin] Admin status panel feature

### PR DESCRIPTION
**Pull request by panigrc**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Selection_002

Selection_003

This feature adds a new Menu in the Administration Panel. The status page shows information about PHP version about Sylius version and enabled PHP modules.
Also there is an `AdministratorStatusMenuBuilderEvent` so you can add buttons to it.

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Event;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Sylius\Bundle\UiBundle\Menu\Event\MenuBuilderEvent;
+
+class AdministratorStatusMenuBuilderEvent extends MenuBuilderEvent
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php]:
```suggestion
final class AdministratorStatusMenuBuilderEvent extends MenuBuilderEvent
```

File: src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+    public function __construct(
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php]: Could fit one line

File: src/Sylius/Bundle/AdminBundle/Menu/AdministratorStatusMenuBuilder.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Menu;
+
+use Knp\Menu\FactoryInterface;
+use Knp\Menu\ItemInterface;
+use Sylius\Bundle\AdminBundle\Event\AdministratorStatusMenuBuilderEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+final class AdministratorStatusMenuBuilder
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Menu/AdministratorStatusMenuBuilder.php]: Why do we need this class?

File: src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php (hunk 1 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php]: Why do we need this class?

Comment (panigrc) [reply src/Sylius/Bundle/AdminBundle/Event/AdministratorStatusMenuBuilderEvent.php]: This class adds a Menu Builder Event to the Administrator status page so someone could add buttons like `Clear cache`

File: src/Sylius/Bundle/AdminBundle/Menu/AdministratorStatusMenuBuilder.php (hunk 3 above)
Comment (panigrc) [reply src/Sylius/Bundle/AdminBundle/Menu/AdministratorStatusMenuBuilder.php]: This class adds a Menu Builder to the Administrator status page so someone could add buttons like `Clear cache`

**Comment by panigrc** [comment]
The `AdministratorStatusMenuBuilder` and the `AdministratorStatusMenuBuilderEvent` classes add the functionality of extending the Menu by adding Buttons in the future.

You could add buttons on the upper right corner of the page like **Clear cache**

**Comment by panigrc** [comment]
Well actually the Administrator Status page was requested/built mainly to add buttons to it, for easy accessible actions.
If needed we could expand this page with other statistics which are important to administrators.

**Comment by oallain** [comment]
As for example the list of Sylius Plugins installed?

**Pull request closed without merging**